### PR TITLE
Add bitbanging pullups, use them for w1-gpio

### PR DESCRIPTION
**Pull request by mkj**
Allows parasite power to work, uses module option pullup=1

### CODE REVIEW

**Comment by popcornmix** [comment]
No real objection, but as this is something I don't have the ability to test, it would be nice to get a confirmation that this patch is good from any other users.

**Comment by mkj** [comment]
Yep, I'd like to see some more testing too. In my case I've tried 3x ds18b20s in parasite mode. I know a few people were interested in the feature so give it a few weeks maybe.

File: drivers/w1/w1_int.c [hunk 1]
```diff
@@ -117,19 +117,22 @@ int w1_add_master_device(struct w1_bus_master *master)
...
-	/* While it would be electrically possible to make a device that
-	 * generated a strong pullup in bit bang mode, only hardware that
-	 * controls 1-wire time frames are even expected to support a strong
-	 * pullup.  w1_io.c would need to support calling set_pullup before
-	 * the last write_bit operation of a w1_write_8 which it currently
-	 * doesn't.
-	 */
+
+    /* bitbanging hardware uses bitbang_pullup, other hardware uses set_pullup
+     * and takes care of timing itself */
```
Comment (richo) [inline drivers/w1/w1_int.c]: Inconsistent indents through here.

**Comment by rudiratlos** [comment]
I tried 2 parasitic 18b20p successfully. Currently with my own Software, which is just Turning on the pullup. Whould be nice to have the pullup=1 Option.
I also Developed Some additional Hardware, which requires a turned on internal pullup.

**Comment by mkj** [comment]
It looks like some other people have had success, might be right to merge now?

http://www.raspberrypi.org/phpBB3/viewtopic.php?p=302392#p302392

**Comment by popcornmix** [comment]
I'm not too keen on the whitespace commit, as it doesn't follow kernel conventions:
https://www.kernel.org/doc/Documentation/CodingStyle

You should run
scripts/Lindent

if you want to tidy up some currently messy code

**Comment by mkj** [comment]
Yeah, the whitespace was a bit variable in the w1 dir, I was aiming for a minimal diff. I've run Lindent on the files I touched now.

**Pull request merged**